src/gui/graphsettings.cpp

```diff
@@ -61,7 +68,7 @@ GraphSettings::~GraphSettings()
 
 void GraphSettings::init()
 {
-    Q_ASSERT_X(m_exp->modelPlugin(), "GraphSettings",
+    Q_ASSERT_X(m_exp && m_exp->modelPlugin(), "GraphSettings",
                "tried to init the graph settings for a null model!");
```

Comment:
considering that you now have a constructor to build a `GraphSettings` without an `exp`, this assert will always fail

Reply:
Oops sorry I forgot about testing the asserts. So should we remove this completely or can we test it conditionally (only test for ``m_exp->modelPlugin()`` if ``m_exp`` exists)?

Reply:
([see my comment below)]https://github.com/evoplex/evoplex/pull/40#discussion_r308015755))